src/main/java/sx/blah/discord/handle/impl/obj/Guild.java

```diff
@@ -664,7 +673,28 @@ public IRegion getRegion() {
 	 * @param regionID The voice region.
 	 */
 	public void setRegion(String regionID) {
-		this.regionID = regionID;
+		DiscordClientImpl clientImpl = (DiscordClientImpl) client;
+		REGION_LOCK.lock();
+		try {
+			if (clientImpl.REGIONS.isEmpty()) {
+				RequestBuffer.request(() ->
+						(List<IRegion>) DiscordUtils.getStandardRegions(clientImpl.REQUESTS))
+						.get().stream() // get() is called to so a request can populate the cache
+						.forEach(r -> clientImpl.REGIONS.put(r.getID(), r));
+			}
```

Comment:
I think this logic does not belong here since it's modifying a separate object.

Reply:
Where else would it go?

Reply:
In the client object itself